syz-fuzzer/fuzzer.go

```diff
@@ -235,11 +235,8 @@ func main() {
 		comparisonTracingEnabled: r.CheckResult.Features[host.FeatureComparisons].Enabled,
 		corpusHashes:             make(map[hash.Sig]struct{}),
 	}
-	var gateCallback func()
-	if r.CheckResult.Features[host.FeatureLeakChecking].Enabled {
-		gateCallback = func() { fuzzer.gateCallback(r.MemoryLeakFrames) }
-	}
-	fuzzer.gate = ipc.NewGate(2**flagProcs, gateCallback)
```

Comment:
We need the gate in all cases, even without leak checking.

Reply:
Do you mean: move this back to main?

The code is still there, just in "useBugFrames". ipc.NewGate does nothing if gateCallback is nil? And it seems only if leak checking is enabled is it set to non-nil.

Reply:
Ah, I see, it should work then, just somewhat confusing b/c gate is not a no-op even without callback. It has major role even without leak/race checking, so seeing it being setup in useBugFrames is somewhat misleading.

Reply:
Moved fuzzer.gate  assignment.